task2(Spring-Core)/task_Spring_EPAM/src/main/java/com/example/task_Spring_EPAM/service/TrainerService.java

```diff
@@ -0,0 +1,74 @@
+package com.example.task_Spring_EPAM.service;
+
+import com.example.task_Spring_EPAM.dao.TrainerDAO;
+import com.example.task_Spring_EPAM.entity.Trainer;
+import com.example.task_Spring_EPAM.util.UsernamePasswordGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.Optional;
+
+@Service
+public class TrainerService {
+
+    private static final Logger logger = LoggerFactory.getLogger(TrainerService.class);
+
+    private TrainerDAO trainerDAO;
+
+    @Autowired
+    public void setTrainerDAO(TrainerDAO trainerDAO) {
+        this.trainerDAO = trainerDAO;
+    }
+
+    public void createTrainer(Trainer trainer) {
+        int serialNumber = 0;
+        String baseUsername = UsernamePasswordGenerator.generateUsername(trainer.getFirstName(), trainer.getLastName());
+        String username = baseUsername;
+        while (trainerDAO.findByUsername(username).isPresent()) {
```

Comment:
this part is duplicating in several DAO`s and may be separated in order to reuse